public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts

```diff
@@ -574,3 +577,49 @@ function trackIfEmpty(grid: SceneGridLayout) {
     sub.unsubscribe();
   };
 }
+
+/**
+ * Behavior that will capture currently selected variables and time range and save them to local storage, so that they can be applied when the next dashboard is loaded.
+ */
+function preserveDashboardSceneStateInLocalStorage(scene: DashboardScene) {
+  if (!config.featureToggles.preserveDashboardStateWhenNavigating) {
+    return;
+  }
+
+  return () => {
+    // Skipping saving state for default home dashboard
+    if (!scene.state.uid) {
+      return;
+    }
+    const variables = scene.state.$variables?.state.variables;
```

Comment:
any reason not using UrlSyncManager.getUrlState(scene) here? (I know it might have more state you don't care about but we can filter that out either before save or when we try to use it)